fsrs/src/main/kotlin/com/inout/fsrs/base/Date.kt

```diff
@@ -20,7 +24,34 @@ fun Date.addMinutes(minutes: Int): Date {
  * Calculate the difference in days between two dates.
  * if difference is 0.9 days, it will be rounded to 0.
  */
-fun Date.diff(other: Date): Int {
+fun Date.diffInDays(other: Date): Int {
     val diffInMills = this.time - other.time
     return (diffInMills / (1000 * 60 * 60 * 24)).toInt()
 }
+
+fun Instant.diffInDays(other: Instant): Int {
+    return diffInMillis(other).toInt() / (1000 * 60 * 60 * 24)
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/in-out-team/api-server/pull/11#issuecomment-2362318726>#0b18f5</a><span> - 09/20/2024, 12:05 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The diffInDays function may produce incorrect results for large time differences due to integer overflow
</div>
<div id="fix">
<b>Fix:</b> Use Long instead of Int for the calculation to prevent overflow for large time differences
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -32,3 +32,3 @@ fun Date.diffInDays(other: Date): Int {
 -fun Instant.diffInDays(other: Instant): Int {
 -    return diffInMillis(other).toInt() / (1000 * 60 * 60 * 24)
 +fun Instant.diffInDays(other: Instant): Long {
 +    return diffInMillis(other) / (1000 * 60 * 60 * 24)
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged